### ISSUE TITLE
Correct cost information no longer being displayed

### ISSUE THREAD

**Issue by mohsanjaffery** [awslabs/aws-perspective#99]
## Description

Following an update to the S3 console the Cost bucket replication steps in the [**Set up the cost feature**] section of the [implementation guide](https://docs.aws.amazon.com/solutions/latest/aws-perspective/automated-deployment.html) are no longer valid.  This can lead to **$0** values being displayed in Perspective for cost generating resources.

## How do I know if I am affected?
- You have deployed AWS Perspective after 01 Nov 2020 and have configured the Cost feature with replication
- After 24 hours you are seeing **$0** values when viewing cost generating resources in Perspective
- You are seeing [**Replication status** : FAILED] when viewing a Cost and Usage report in its S3 bucket

### Set up bucket replication

Set up replication to the S3 bucket created during deployment. The S3 bucket is named with the following format: aws-perspective-`<*deployment-accountID*>`-`<*deployment-region*>`-cost-bucket.  This allows AWS Perspective to process the cost and usage data and map it to the resources it has already discovered.  Please perform the following actions to complete replication configuration.

### The account where Cost and Usage reports are configured

    1. Sign in to the Amazon S3 console - https://console.aws.amazon.com/s3/
    2. Select the S3 bucket created when configuring your AWS Cost and Usage Report
    3. Select the [Management] tab, and then choose [Replication rules]
    4. Choose  [Create replication rule]
    5. Within the [Replication rule] configuration section enter a descriptive rule ID for the [Replication rule name]
    6. In the [Source bucket] section select [This rule applies to all objects in the bucket] to configure the rule scope
    7. In the [Destination] section configure the following
        1. Select [Specify a bucket in another account]
        2. Enter the account ID
        3. Enter the value for the [Bucket name] with logical-id [PerspectiveCostBucket]
        4. Select the checkbox next to [Change object ownership to destination bucket owner]
    8. In the [IAM] section
        1. Under [IAM role], choose [Create a new role]
        2. If a replication role already exists select it and ensure it includes the [Replication Role Policy Actions] below
    9. Log into the AWS console where AWS Perspective is installed, navigate to the [S3] service page and select the bucket from [step 3] above
    10. From the destination buckets [Management] tab scroll down to the [Replication rules] section and select [Receive replicated objects] from the [Actions] drop-down menu
    11. Within the [Source bucket account settings] section
        1. Add a value for the [Source bucket account ID]
        2. Select [Generate policies]
        3. From the newly created [Policies] section expand the [Bucket policy]
        4. Select the option to [Include permission to change object ownership to destination bucket owner]
        5. Select [Copy] to grab the generated policy

### The account where AWS Perspective is deployed

    1. Navigate to the [S3] service page and find the bucket with the logical ID of PerspectiveCostBucket
    2. Select the [Permissions] tab, scroll down to the [Bucket policy] section and click on [edit]
    3. Paste the previously copied policy and click on [Save changes]

When the reports are in the AWS Perspective account you will start to see cost data appearing on the bounding boxes and individual resources

## <a name="repl-actions"></a>Replication Role Policy Actions

| IAM Policy Action | Description |
| --- | --- |
| s3:ReplicateObject | Grants permission to replicate objects and object tags to the destination bucket |
| s3:ReplicateDelete | Grants permission to replicate delete markers to the destination bucket |
| s3:ReplicateTags | Grants permission to replicate object tags to the destination bucket |
| s3:ObjectOwnerOverrideToBucketOwner | Grants permission to change replica ownership |
| s3:ListBucket | Grants permission to list some or all of the objects in an Amazon S3 bucket |
| s3:GetReplicationConfiguration | Grants permission to get the replication configuration information set on an Amazon S3 bucket |
| s3:GetObjectVersionForReplication | Grants permission to replicate both unencrypted objects and objects encrypted with SSE-S3 or SSE-KMS |
| s3:GetObjectVersionAcl | Grants permission to return the access control list (ACL) of a specific object version |
| s3:GetObjectVersionTagging | Grants permission to return the tag set for a specific version of the object |
| s3:GetObjectRetention | Grants permission to retrieve the retention settings for an object |
| s3:GetObjectLegalHold | Grants permission to get an object's current Legal Hold status |

**Comment by tamaldatta91** [comment]
Yes, I am also seeing the same cost issue. It's not showing correct cost.
